### ISSUE TITLE
iOS apps

### ISSUE THREAD

**Issue by bohwaz** [kd2org/karadav#22]
It seems that iOS NextCloud and ownCloud apps don't work, each with different bugs.

I don't have any Apple device, and I'm not really interested, but if anyone is interested to try and fix the server for those clients, feel free :)

**Comment by tebra-jl** [comment]
Hi,
Everything works as expected with web UI. Display and manage all my files.
This is the « about » informations in Nextcloud client on my ipad:
Nextcloud Liquid for iOS 4.7.0.23 @ 2023
Nexcloud Server 24.0.4
(null) - (null)

I can login to my karadav v 0.3.10 server with that client but there is no files and directory displayed.

**Comment by bohwaz** [comment]
Yes as said, iOS apps don't work, but I cannot do anything as I don't have any Apple device, someone else with an Apple device and some developer knowledge should be able to find the issue, but I can't help.

**Comment by bohwaz** [comment]
Yes, any other webdav app should work.

**Comment by xor-gate** [comment]
I have tested with Nextcloud iOS app 33.0.2 and karadav https://github.com/kd2org/karadav/commit/7cb199e59265038c7e391fc16615d342722496d5 and the browser popup seems to work but after login there is no succesfull redirect back to the app and it is not coupled. I did not further debug it.

**Comment by poVoq** [comment]
I think version later than 0.6.3. are currently broken with Nextcloud clients.

**Comment by xor-gate** [comment]
Native Nextcloud Desktop Client Version 4.0.6 (macOS) works fine on my iMac!

**Comment by bohwaz** [comment]
I think that macOS was always fine, the issue is specifically with iOS apps.

**Comment by adamshand** [comment]
It's really rough, but if anyone wants to try it out I have a fork of KaraDAV that works with OwnCloud and NextCloud on macOS and iOS.  Only tested on latest version of everything (macOS, iOS, NextCloud, OwnCloud and KaraDAV)

OwnCloud support was simple, and seems to work well.  

NextCloud was a bit more involved, but it supports all features I'm aware of including:
- Virtual files syncing on macOS with Finder integration
- Media and Activity view on iOS.

I'm sure there are bugs, but it seems to "work for me".  I'll keep testing and happy to send a PR in a week or two if that's useful.

https://github.com/adamshand/karadav

**Comment by bohwaz** [comment]
Yes I would be happy to receive PRs for fixes, but please explain what is done (if not obvious) to help me understand :)

**Comment by bohwaz** [comment]
Hi @adamshand thanks for your commits I took a look and implemented most of the changes, and posted comments on other stuff that seems… big. I'm trying to have minimal support (file sync, thumbnails), not complete support of mobile apps, to keep the code light and easy to maintain.

I'm also trying to keep most of the NextCloud code outside of WebDAV/Server as it is used in other software that don't implement NextCloud compatibility.

Feel free to add comments and suggest PRs.

**Comment by adamshand** [comment]
Hey cool, thanks!  Glad it was useful and thanks for merging!

Do you want me to test your changes since I don't think you have iOS/macOS for testing?

Sorry, this is mostly vibe slop as I was experimenting and trying to figure out how to get things working.  Was gonna clean it up before I sent though a PR!

Totally understand wanting to keep things simple and easy to maintain.  

I'll keep maintaining a fork for the extra stuff.  Maybe over time can figure out nice ways to contribute it back which make sense for you, but right now it hasn't even been tested enough for me to be confident it all works properly.

**Comment by raelovejoy** [comment]
ownCloud for iOS works with my installation of KaraDAV behind NGINX Proxy Manager:

```
ownCloud release version 12.7.0 build 302 (Wed May  6 13:16:46 UTC 2026)
(app: v12.7.0 - @3fac7ea, sdk: v12.4.0-71-g275bad5a - @275bad5a)

```

Here's my NGINX custom configuration:

```
# KaraDAV compatibility shims for ownCloud/OpenCloud clients.
# Keep /status.php complete because some clients require productversion.

location = /status.php {
    default_type application/json;
    return 200 '{
        "installed": true,
        "maintenance": false,
        "needsDbUpgrade": false,
        "version": "10.15.2.0",
        "versionstring": "10.15.2",
        "productversion": "10.15.2",
        "edition": "",
        "productname": "",
        "extendedSupport": false
    }';
}

# ownCloud Desktop rejects KaraDAV's root href when it lacks the trailing slash.
# Rewrite only that exact root href in XML responses.
location = /remote.php/webdav/ {
    proxy_pass [IP of your KaraDAV];

    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto https;
    proxy_set_header X-Forwarded-Host $host;

    sub_filter_types application/xml text/xml;
    sub_filter_once off;
    sub_filter '<d:href>/remote.php/webdav</d:href>' '<d:href>/remote.php/webdav/</d:href>';
}

```

**Comment by bohwaz** [comment]
Hi, thanks for your feedback, please test and report if 0.7.0 beta 4 works with iOS clients or not :) Both issues reported by @raelovejoy should be OK.
